docs/DeveloperGuide.md

```diff
@@ -1,37 +1,293 @@
+
+**Step3.** `printList` will first check if the size of the private ArrayList named `projectList` is 0, which means that the arraylist is empty. If yes, it will print "You have not added any projects yet!" and return.
+
+**Step4.** It will then iterate over `projectList`, which type is `Project`. It will call `getTitle` and `getDeadline` from each array element in`the `Project` class.
+
+**Step5.** the general information of a project, i.e., its title and deadline, will be displayed.
+
+#### Todo feature
+![image info](./UmlDiagrams/UpdatedTodoSequence.png)  
+
+**Step1.** When `CommandHandler` receives a user input starting with string “todo”, it will create a `AddTodoCommandParser` object and call `parse()`function to parse the user input.  
+
+**Step2.** The `AddTodoCommandParser` object will parse the todo input. Upon parsing the user input, it will create a `AddTodoCommand` and return it back to `CommandHandler`.  
+
+**Step3.** After `CommandHandler` obtains the `AddTodoCommand` returned by `AddTodoCommandParser`, it then calls `executeCommand()` method in its own class.  
+
+**Step4.** `executeCommand()` will call `execute()` method of `AddTodoCommand`. `AddTodoCommand` will call `addTodoToProject()` method in order to get the target project and add todo to it.
+
+#### View a Project
+
+The view function makes use of ```ProjectList``` and ```CommandHandler``` classes. It is facilitated by methods within the ```Todo``` and ```Project``` classes, stored within ```ProjectList```.
+Given below is an example usage scenario and how View Project behaves at each step
+
+**Step 1.** The user executes ```view cs2113``` to view the project titled cs2113. The command is retrieved and parsed by ```handleUserInput()``` in ```CommandHandler```, which calls ```view()``` in ```CommandHandler```
+
+**Step 2.** The ```view()``` method ensures that ```ProjectList``` has been correctly initialised and the command is valid, before calling ```view()``` in ```ProjectList```
+
+**Step 3.** ```view()``` in ```ProjectList``` calls ```findProjectIndex()``` in ```ProjectList```
+
+**Step 4.** ```findProjectIndex()``` loops through all ```Project``` objects in projectList until a ```Project``` with the same title as the one entered by the user is found. The ```Project```’s index is returned to ```CommandHandler```'s implementation of ```view()```
+
+**Step 5.** ```view()``` calls the ```printDetails()``` method of the ```Project``` at the given index
 
-## Design & implementation
+**Step 6.** ```printDetails()``` displays project title and deadline to the user. It iterates through the ```Todo```s in the ```todos``` attribute and prints each `Todo` by calling `Todo`’s `toString()` function
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![image info](./UmlDiagrams/ViewProject.jpg)
```

Comment:
Shouldn't arrow heads be at the tip of the activation bars?
Also, for the alt statement, wouldn't it be better to use an opt since there is no else clause?